### ISSUE TITLE
Can't delete Jobs submitted

### ISSUE THREAD

**Issue by ARWattam** [PATRIC3/patric3_website#398]
It would be nice if we could delete these, especially if you made a mistake with one and need to add more genomes.  Can't get rid of the older version

**Comment by mshukla1** [comment]
I think there is a more general issue here: ability to delete jobs and corresponding results in the workspace and/or in the database.

**Comment by olsonanl** [comment]
Yes, definitely. I know I have a ton of junk genomes in the database and ws because of testing.

**Comment by nconrad** [comment]
@olsonanl, @mshukla1 was talking about this today.   I think the 'delete' and 'suspend' from https://github.com/MG-RAST/AWE/wiki/AWE-APIs would need to be made available via the app_service, right?  It seems the safest way to delete the corresponding data would be to use their hashes?  So then app service would need to keep track of those and the workspaces may need to accept hashes for deletion (if it doesn't already).

**Comment by olsonanl** [comment]
Yes. We’d need to also verify that they actually kill the jobs. I don’t necessarily want to delete the actual data, since we use that for our reporting. We may need to have a flag-deleted or somesuch.

At some point I am going to have the app service maintain its own database of this data; this is required for moving away from AWE.

This is another thing that isn’t terribly difficult but needs to be done with care.

—bob

On Sep 14, 2017, at 12:40 PM, nconrad <notifications@github.com<mailto:notifications@github.com>> wrote:


@olsonanl<https://github.com/olsonanl>, @mshukla1<https://github.com/mshukla1> was talking about this today. I think the 'delete' and 'suspend' from https://github.com/MG-RAST/AWE/wiki/AWE-APIs would need to be made available via the app_service, right? It seems the safest way to delete the corresponding data would be to use their hashes? So then app service would need to keep track of those and the workspaces may need to accept hashes for deletion (if it doesn't already).

**Comment by nconrad** [comment]
I'm happy to say you can delete jobs now.  Thanks to Bob.

**Comment by mshukla1** [comment]
@nconrad 

How? 

I know I can kill a job that is queued or running. However, I still can't delete a already completed job from my jobs list...and corresponding results, assuming they are still in the same place as originally specified output folder.

**Comment by nconrad** [comment]
@mshukla1 Doh, yes, you are right.

**Comment by nconrad** [comment]
This ticket needs review.  I think it is a dangerous slippery slope to delete underlining data when you delete job information.  Is that what is wanted?
